Sources/GRPCProtobufCodeGen/CamelCaser.swift

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package struct CamelCaser {
+  /// Converts a string from upper camel case to lower camel case.
+  package static func toLowerCamelCase(_ s: String) -> String {
+    if s.isEmpty { return "" }
```

Comment:
I don't think we need this check. If the string is empty then `indexOfFirstLowerCase` will be `nil` and we'll return `s.lowercased()` (which is fine)